project/account/models.py

```diff
@@ -0,0 +1,101 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, username, first_name, last_name, password=None):
+        """
+        Creates and saves a user with the given email, username and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+        if not username:
+            raise ValueError("Users must have a username")
+        if not first_name:
+            raise ValueError("Users must have a first name")
+        if not last_name:
+            raise ValueError("Users must have a last name")
+        if not password:
+            raise ValueError('Users must have a password')
+
+        user = self.model(
+            email=self.normalize_email(email),
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+        )
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email, username, first_name, last_name, password):
+        """
+        Creates and saves a staff user with the given email, username and password.
+        """
+        user = self.create_user(
+            email=self.normalize_email(email),
```

Comment:
You don't have to use "self.normalize_email(email)" again. Since you are calling the "create_user" method where you used it initially.

Reply:
email=self.normalize_email(email): This is used to clean up the email, to ensure it follows the right format. It is a check on the models.EmailField table(or is it row?)